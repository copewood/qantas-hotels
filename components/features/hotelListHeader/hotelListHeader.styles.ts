import styled from 'styled-components';

export const StyledHeaderContainer = styled.div`
  width: 100%;
  flex: 1;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`;

export const StyledImageContainer = styled.img`
  width: 200px;
  margin-bottom 1rem;
`;

export const StyledSelect = styled.select`
  width: 130px;
  margin-left 1rem;
`;