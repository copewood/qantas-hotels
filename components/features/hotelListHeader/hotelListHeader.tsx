import React, { FC } from 'react';
import { Hotel } from '@components/models/hotel';

import { 
  StyledHeaderContainer, 
  StyledImageContainer,
  StyledSelect
} from './hotelListHeader.styles';

type HotelListHeaderProps = {
  hotels: Array<Hotel>
  setSortOrder: (sortOrder: string) => void;
};

export const HotelListHeader: FC<HotelListHeaderProps> = ({hotels, setSortOrder}) => {

  const sortList = (event) => setSortOrder(event.currentTarget.value);

  return <>
    <StyledImageContainer src="/qantas-logo.png" />

    <StyledHeaderContainer>
      <div>
        {hotels.length} hotels in Sydney
      </div>
      <div>
        Sort by
        <StyledSelect onClick={sortList} name="sort" id="sort">
          <option value="desc">Price high-low</option>
          <option value="asc">Price low-high</option>
        </StyledSelect>
      </div>
    </StyledHeaderContainer>
  </>;
};
