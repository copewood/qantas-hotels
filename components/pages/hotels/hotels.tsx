import React, { FC, useEffect, useState} from 'react';

import { HotelListHeader } from 'components/features/hotelListHeader/hotelListHeader';
import { HotelList } from 'components/features/hotelList/hotelList';
import { Hotel } from '@components/models/hotel';

type HotelsProps = {};

const fetchHotels = async () => {
    return fetch('/api/hotels')
        .then((response) => response.json())
        .then((responseJson) => responseJson);
}

export const Hotels: FC<HotelsProps> = ({}) => {
  const [hotels, setHotels] = useState<Array<Hotel>>();

  useEffect(() => {
    fetchHotels().then((response)=>{
      setHotels(response.results as Array<Hotel>)
    });
  },[])
  
  return <>
    {hotels && <>
      <HotelListHeader hotels={hotels}/>
      {/* <HotelList hotels={hotels}/> */}
    </>}
  </>;
};
