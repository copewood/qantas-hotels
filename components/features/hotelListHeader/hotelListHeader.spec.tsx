import React from 'react';
import { render } from '@testing-library/react';
import { HotelListHeader } from './hotelListHeader';
import { mockHotels } from '../../../pages/api/hotels'

const mockSetSortOrder = jest.fn();

describe('HotelList', () => {
  it('should show HotelList', () => {
    const { container } = render(<HotelListHeader hotels={mockHotels.results} setSortOrder={mockSetSortOrder} />);
    expect(container).toMatchSnapshot()
  });
});
