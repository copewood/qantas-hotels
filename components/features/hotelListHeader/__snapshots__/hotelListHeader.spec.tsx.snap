// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HotelList should show HotelList 1`] = `
<div>
  <img
    class="sc-gsnTZi eToNsf"
    src="/qantas-logo.png"
  />
  <div
    class="sc-bczRLJ la-DgbC"
  >
    <div>
      5
       hotels in Sydney
    </div>
    <div>
      Sort by
      <select
        class="sc-dkzDqf gKtHBM"
        id="sort"
        name="sort"
      >
        <option
          value="desc"
        >
          Price high-low
        </option>
        <option
          value="asc"
        >
          Price low-high
        </option>
      </select>
    </div>
  </div>
</div>
`;
